test/unit/BCoWHelper.t.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.25;
+
+import {BCoWHelper} from 'contracts/BCoWHelper.sol';
+import {Test} from 'forge-std/Test.sol';
+
+import {IBCoWPool} from 'interfaces/IBCoWPool.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
+import {ISettlement} from 'interfaces/ISettlement.sol';
+
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+
+import {ICOWAMMPoolHelper} from '@cow-amm/interfaces/ICOWAMMPoolHelper.sol';
+import {GPv2Interaction} from '@cowprotocol/libraries/GPv2Interaction.sol';
+import {GPv2Order} from '@cowprotocol/libraries/GPv2Order.sol';
+
+import {MockBCoWFactory} from 'test/manual-smock/MockBCoWFactory.sol';
+import {MockBCoWPool} from 'test/manual-smock/MockBCoWPool.sol';
+
+contract BCoWHelperTest is Test {
+  BCoWHelper helper;
+
+  MockBCoWFactory factory;
+  MockBCoWPool pool;
+  address invalidPool = makeAddr('invalidPool');
+  address[] tokens = new address[](2);
+  uint256[] priceVector = new uint256[](2);
+
+  uint256 constant VALID_WEIGHT = 1e18;
+
+  function setUp() external {
+    factory = new MockBCoWFactory(address(0), bytes32(0));
+
+    address solutionSettler = makeAddr('solutionSettler');
+    vm.mockCall(
+      solutionSettler, abi.encodePacked(ISettlement.domainSeparator.selector), abi.encode(bytes32('domainSeparator'))
+    );
+    vm.mockCall(
+      solutionSettler, abi.encodePacked(ISettlement.vaultRelayer.selector), abi.encode(makeAddr('vaultRelayer'))
+    );
+    pool = new MockBCoWPool(makeAddr('solutionSettler'), bytes32(0));
+
+    // creating a valid pool setup
+    factory.mock_call_isBPool(address(pool), true);
+    tokens[0] = makeAddr('token0');
+    tokens[1] = makeAddr('token1');
+    pool.set__tokens(tokens);
+    pool.set__records(tokens[0], IBPool.Record({bound: true, index: 0, denorm: VALID_WEIGHT}));
+    pool.set__records(tokens[1], IBPool.Record({bound: true, index: 1, denorm: VALID_WEIGHT}));
+    pool.set__totalWeight(2 * VALID_WEIGHT);
+    pool.set__finalized(true);
+
+    priceVector[0] = 1e18;
```

Comment:
I'd recommend using a price ratio that's not 1:1 so the unit tests can detect if we invert the order of parameters, for example

Reply:
i did! `1:1.05`